lib/zlib.js

```diff
@@ -251,9 +251,20 @@ function Zlib(opts, mode) {
       opts.finishFlush, 'options.finishFlush',
       Z_NO_FLUSH, Z_BLOCK, Z_FINISH);
 
-    windowBits = checkRangesOrGetDefault(
-      opts.windowBits, 'options.windowBits',
-      Z_MIN_WINDOWBITS, Z_MAX_WINDOWBITS, Z_DEFAULT_WINDOWBITS);
+      // windowBits is special. On the compression side, 0 is an invalid value.
+      // But on the decompression side, a value of 0 for windowBits tells zlib
+      // to extract the value from the headers of the compressed stream.
+    if (opts.windowBits === 0 &&
+        (mode === INFLATE ||
+         mode === GUNZIP ||
+         mode === INFLATERAW ||
+         mode === UNZIP)) {
+      windowBits = 0
```

Comment:
Can you run `make lint`? It should complain about the lack of a semicolon here ;)

Reply:
whoops. I'm spoiled by standardjs! Will fix.